packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts

```diff
@@ -0,0 +1,155 @@
+  { postId: 'JPan54R525D68NoEt', sequenceId: '5Eg2urmQjA4ZNcezy' },
+  { postId: 'N5Jm6Nj4HkNKySA5Z', sequenceId: 'kxs3eeEti9ouwWFzr' },
+  { postId: 'NQgWL7tvAPgN2LTLn', sequenceId: '6eWnDWYn5Nudg2k7L' },
+  { postId: 'NxF5G6CJiof6cemTw', sequenceId: '4dHMdK5TLN6xcqtyc' },
+  { postId: 'P6fSj3t4oApQQTB7E', sequenceId: 'xEFeCwk3pdYdeG2rL' },
+  { postId: 'p7x32SEt43ZMC9r7r', sequenceId: 'Rm6oQRJJmhGCcLvxh' },
+  { postId: 'rz73eva3jv267Hy7B', sequenceId: 'akMLzwcRdJNnmBoLa' },
+  { postId: 'rzqACeBGycZtqCfaX', sequenceId: '5Eg2urmQjA4ZNcezy' },
+  { postId: 'SwcyMEgLyd4C3Dern', sequenceId: 'HeYtBkNbEe7wpjc6X' },
+  { postId: 'wEebEiPpEwjYvnyqq', sequenceId: 'xEFeCwk3pdYdeG2rL' },
+  { postId: 'wQACBmK5bioNCgDoG', sequenceId: 'pC6DYFLPMTCbEwH8W' },
+  { postId: 'xCxeBSHqMEaP3jDvY', sequenceId: '7CdoznhJaLEKHwvJW' },
+  { postId: 'xJyY5QkQvNJpZLJRo', sequenceId: 'HmANELvkhAZ9eDxFS' },
+  { postId: 'YABJKJ3v97k9sbxwg', sequenceId: 'xEFeCwk3pdYdeG2rL' },
+  { postId: 'yeADMcScw8EW9yxpH', sequenceId: 'zucjLBpQ9S9eWPWGu' },
+  { postId: 'YN6daWakNnkXEeznB', sequenceId: 'hBFDRZCPLcrRDubgm' },
+  { postId: 'zB4f7QqKhBHa5b37a', sequenceId: 'CmrW8fCmSLK7E25sa' },
+  { postId: 'ZDZmopKquzHYPRNxq', sequenceId: 'NkCAHsDPchBHabEuc' },
+  { postId: 'zp5AEENssb8ZDnoZR', sequenceId: 'yiFxBWDXnLpbWGTkK' },
+  { postId: 'zTfSXQracE7TW8x4w', sequenceId: 'uLEjM2ij5y3CXXW6c' },
+  { postId: '5gfqG3Xcopscta3st', sequenceId: 'ZbmRyDN8TCpBTZSip' },
+  { postId: 'rYJKvagRYeDM8E9Rf', sequenceId: 'pC6DYFLPMTCbEwH8W' },
+  { postId: '7im8at9PmhbT4JHsW', sequenceId: 'n945eovrA3oDueqtq' },
+  { postId: 'ax695frGJEzGxFBK4', sequenceId: 'n945eovrA3oDueqtq' },
+  { postId: 'nNqXfnjiezYukiMJi', sequenceId: 'n945eovrA3oDueqtq' },
+  { postId: 'RQpNHSiWaXTvDxt6R', sequenceId: 'FYMiCeXEgMzsB5stm' },
+];
+
+registerMigration({
+  name: 'undoBestOfCollectionClobbering',
+  dateWritten: '2024-10-31',
+  idempotent: true,
+  action: async () => {
+    const collectionId = 'nmk3nLpQE89dMRzzN'; 
+
+    // Step 1: Remove canonical fields from affected posts
+    const collectionDoc = await Collections.findOne({ _id: collectionId });
+    if (!collectionDoc) {
+      // eslint-disable-next-line no-console
+      console.error(`Collection with ID ${collectionId} not found.`);
+      return;
+    }
+    const collectionSlug = collectionDoc.slug;
+
+    const postsToUnset = await Posts.find({ canonicalCollectionSlug: collectionSlug }).fetch();
+    // eslint-disable-next-line no-console
+    console.log(`Found ${postsToUnset.length} posts to unset canonical fields.`);
+
+    if (postsToUnset.length > 0) {
+      const updates = postsToUnset.map(post => ({
+        updateOne: {
+          filter: { _id: post._id },
+          update: {
+            $unset: {
+              canonicalPrevPostSlug: "",
+              canonicalNextPostSlug: "",
+              canonicalBookId: "",
+              canonicalCollectionSlug: "",
+              canonicalSequenceId: "",
```

Comment:
Huh, surprised that this is the syntax for bulk unset operations.  (Thought it does match up to other uses.)

Reply:
I questioned it too (as in questioned the model about the empty strings)